### PR TITLE
Template Literal handling

### PR DESCRIPTION
**Pull request by tvillaren**
# Why

This PR implement supports for #64 and #214 

### Issue #64

Transforms 
```ts
export enum Foo {
  Bar = 'Bar',
  Baz = 'Baz',
}

export interface Party {
  Time: `${Foo}`;
}
```
into
 
```ts
export const fooSchema = z.nativeEnum(Foo);
export const partySchema = z.object({
  Time: z.union([z.literal('Bar'), z.literal('Baz')]),
});
```

### Issue #214 
Transforms
```ts
export type Prefix = "large" | "high";
export type Subfix = "back" | "front";
export type Joint = `${Prefix}-${Subfix}`;
```
into 
```ts
export const PrefixSchema = z.union([
  z.literal("large"),
  z.literal("high"),
]);

export const SubfixSchema = z.union([
  z.literal("back"),
  z.literal("front"),
]);

export const JointSchema =  z.union([
  z.literal("large-back"),
  z.literal("large-front"),
  z.literal("high-back"),
  z.literal("high-front"),
]);
```

### CODE REVIEW

**Comment by schiller-manuel** [comment]
Does it make sense to add tests to cover the unhit branches?
e.g. https://app.codecov.io/gh/fabien0102/ts-to-zod/pull/217/blob/src/core/generateZodSchema.ts#L1010

File: src/utils/generateCombinations.test.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
+import { generateCombinations } from "./generateCombinations";
+
+describe("generateCombinations", () => {
+  it("should generate all combinations of 1 empty array", () => {
+    const arrays = [[]];
+    const result = generateCombinations(arrays);
+    expect(result).toEqual([""]);
```
Comment (schiller-manuel) [inline src/utils/generateCombinations.test.ts:7]: why is the result not an empty array `[]`?

**Comment by tvillaren** [comment]
> Does it make sense to add tests to cover the unhit branches? e.g. https://app.codecov.io/gh/fabien0102/ts-to-zod/pull/217/blob/src/core/generateZodSchema.ts#L1010

I don't think it does, other warnings in the lib are not covered either (and I believe those are the only non-covered branches)

File: src/utils/generateCombinations.test.ts (hunk 1 above)
Comment (tvillaren) [reply src/utils/generateCombinations.test.ts]: No reason, it would probably be more logical to be empty indeed. 
Updated accordingly.

**Comment by schiller-manuel** [comment]
LGTM
let's merge!

**Pull request merged**